animals/animal.go

```diff
@@ -0,0 +1,50 @@
+package animals
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type Species int
+
+var savedAnimals []Animal
+
+const (
+	Pig Species = iota
+	Horse
+	Cow
+)
+
+var SpeciesNames = [3]string{
+	Pig:   "Pig",
+	Horse: "Horse",
+	Cow:   "Cow",
+}
+
+type Animal struct {
+	Species Species
+	Name    string
+	Age     int
+}
+
+func IndexHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	encoder := json.NewEncoder(w)
```

Comment:
Yup, can't get much simpler than that ;) You can remove the intermediate variable and do:

```
json.NewEncoder(w).Encode(savedAnimals)
```